src/auth/googleauth.ts

```diff
@@ -243,34 +244,11 @@ export class GoogleAuth {
    * @returns A promise that resolves with the boolean.
    * @api private
    */
-  async _checkIsGCE(isRetry = false): Promise<boolean> {
+  async _checkIsGCE(): Promise<boolean> {
     if (this.checkIsGCE !== undefined) {
       return this.checkIsGCE;
     }
-    if (!this.transporter) {
-      this.transporter = new DefaultTransporter();
-    }
-    try {
-      const res = await this.transporter.request(
-          {url: 'http://metadata.google.internal'});
-      this.checkIsGCE =
-          res && res.headers && res.headers['metadata-flavor'] === 'Google';
-    } catch (e) {
-      const isDNSError = (e as NodeJS.ErrnoException).code === 'ENOTFOUND';
-      const ae = e as AxiosError;
-      const is5xx = ae.response &&
-          (ae.response.status >= 500 && ae.response.status < 600);
-      if (is5xx) {
-        // Unexpected error occurred. Retry once.
-        if (!isRetry) {
-          return await this._checkIsGCE(true);
-        }
-        throw e;
-      } else if (!isDNSError) {
-        throw e;
-      }
-      this.checkIsGCE = false;
-    }
+    this.checkIsGCE = await gcpMetadata.isAvailable();
```

Comment:
:+1: on getting rid of so much duplicated functionality. I think the retry handling is subtly different (I think 5xx errors used to retry only once). Should be fine though.